akka-actor/src/main/scala/akka/dispatch/Mailboxes.scala

```diff
@@ -182,10 +183,12 @@ private[akka] class Mailboxes(
       mailboxType
     }
 
-    if (deploy.mailbox != Deploy.NoMailboxGiven) {
+    if (deploy.mailbox != DefaultTypedMailboxId && deploy.mailbox != Deploy.NoMailboxGiven) {
```

Comment:
The mailbox selector priority is now as follows:

1. Custom mailbox
2. Custom dispatcher with mailbox configuration
3. Default typed actor mailbox
4. The originals...

By changing the order and using a redundant checks, Typed Actors and Classic Actors now has the same behavior on a custom dispatcher.